copy.go

```diff
@@ -21,19 +21,50 @@ import (
 	"fmt"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/semaphore"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
 	"oras.land/oras-go/v2/internal/cas"
 	"oras.land/oras-go/v2/internal/graph"
 	"oras.land/oras-go/v2/internal/status"
 )
 
+// defaultConcurrency is the default concurrency limit.
+// The value is consistent with dockerd and containerd.
+const defaultConcurrency = int64(3)
+
+// CopyOptions contains parameters for oras.Copy.
+type CopyOptions struct {
+	CopyGraphOptions
```

Comment:
`ReferenceFetcher` and `ReferencePusher` can only be applied to the root node. Maybe we can have some options for clients to use them in `Copy()` and do something like:
1. Fetch the root by reference
2. Call `CopyGraph()` on the down edges of the root
3. Push the root by reference

So that we can save 1 `Resolve()` call and 1 `Tag()` call.